mkosi/__init__.py

```diff
@@ -1873,111 +1873,97 @@ def make_rpm_list(args: MkosiArgs, packages: Set[str], do_run_build_script: bool
     return packages
 
 
-def clean_dnf_metadata(root: Path, always: bool) -> None:
-    """Remove dnf metadata if /bin/dnf is not present in the image
+def flatten(lists: Iterable[Iterable[T]]) -> Sequence[T]:
+    """Flatten a sequence of sequences into a single sequnce (list)."""
+    return list(itertools.chain.from_iterable(lists))
 
-    If dnf is not installed, there doesn't seem to be much use in
-    keeping the dnf metadata, since it's not usable from within the
-    image anyway.
-    """
-    paths = [
-        root / "var/lib/dnf",
-        *root.glob("var/log/dnf.*"),
-        *root.glob("var/log/hawkey.*"),
-        root / "var/cache/dnf",
-    ]
 
-    cond = always or not os.access(root / "bin/dnf", os.F_OK, follow_symlinks=False)
+def clean_paths(
+        root: Path,
+        globs: Sequence[str],
+        tool: str,
+        always: bool) -> None:
+    """Remove globs under root if always or if tool is not found under root."""
 
-    if not cond or not any(path.exists() for path in paths):
+    tool = root / tool.lstrip('/')
+    cond = always or not os.access(tool, os.F_OK, follow_symlinks=False)
+
+    paths: Sequence[Path] = flatten(root.glob(glob.lstrip('/')) for glob in globs)
+
+    if not cond or not paths:
         return
 
-    with complete_step("Cleaning dnf metadata…"):
+    with complete_step(f"Cleaning {tool.name} metadata…"):
```

Comment:
`tool` is `str`, not `Path` so should just be `with complete_step(f"Cleaning {tool} metadata…"):`

Reply:
Here, `tool` is a `Path`. This way we get a nice message with the last component only, not the full path.